File: 프로그래머스/피로도.py
from itertools import permutations
# 피로도 시스템(정수) 최소피로도 소모 피로도
def solution(k, dungeons):
    #answer = -1
    count = 0
    c = list(permutations(dungeons,len(dungeons)))

    #k >= dungeons[0][0] -> k -= dungeons[0][1]
    #n = len(dungeons)
    for i in c:
        temp = 0
        temp_k = k
        for current, use in i:
            if temp_k >= current:
                temp_k -= use
                temp += 1   
        #print(count,temp)
        count = max(count,temp)
        
    return count

File: 프로그래머스/피로도.java
class Solution {  
    public int dfs(int depth, int fatigue, int[][] dungeons, boolean[] visited) { 
        int count = depth; // 현재 깊이를 카운트
        for (int i = 0; i < dungeons.length; i++){  
            if (visited[i] || dungeons[i][0] > fatigue) {  
                continue;  
            }  
            visited[i] = true;  
            count = Math.max(count, dfs(depth + 1, fatigue - dungeons[i][1], dungeons, visited));  
            visited[i] = false;  
        }  
        return count; // 최대 깊이를 반환
    }  
    
    public int solution(int k, int[][] dungeons) {  
        boolean[] visited = new boolean[dungeons.length];  
        return dfs(0, k, dungeons, visited); // dfs의 결과를 반환
    }  
}
